AIS/Scrapper.py

```diff
@@ -0,0 +1,101 @@
+                else:
+                    if '\xa0' in d:
+                        self.data['ids'][f'{self.ids[length]}'][f"{H[i]}"] = d.replace('\xa0','').strip()
+                    else:
+                        self.data['ids'][f'{self.ids[length]}'][f"{H[i]}"] = d
+            #extract link for vessel data
+            link = f.find('a', attrs={'href': re.compile("^https://")})
+            if self.data['ids'][f'{self.ids[length]}']["Ships"] == '0':
+                self.data['ids'][f'{self.ids[length]}']['Vessels'] = None
+            else:
+                self.data['ids'][f'{self.ids[length]}']['Vessels'] = {"Status_URL":f"{link.get('href')}"+"?page=","MMSI":{}}
+            self.data['ids'][f'{self.ids[length]}']['Updated_Time'] = datetime.now().strftime("%d/%m/%Y %H:%M:%S")
+            return self.scrape_data(length+1)
+            
+    def vessel_data_scrap(self,url,id,length=1):
+        """
+            Extracting Details of vessels avaliable online location and id specified. 
+        """
+        page = BeautifulSoup(requests.get(url+str(length)).content,'lxml')
+        if length == (int(int(page.find('div', attrs={ 'class' : 'summary'}).findAll('b')[1].getText())/8)+1):
+            return self.data
+        else:
+            for r in range(len(page.findChild('tbody').findAll('tr'))):
+                td = list(tsd.getText() for tsd in page.findChild('tbody').findAll('tr')[r].findAll('td'))
+                for i in range(len(td)):
+                    if i == 0:
+                        self.data['ids'][f'{id}']['Vessels']['MMSI'][f'{td[0]}'] = {}
+                    elif i == 1:
+                        self.data['ids'][f'{id}']['Vessels']['MMSI'][f'{td[0]}']['Name'] = td[i]
+                    elif i == 2:
+                        self.data['ids'][f'{id}']['Vessels']['MMSI'][f'{td[0]}']['Distinct'] = td[i]
+                    elif i == 3:
+                        self.data['ids'][f'{id}']['Vessels']['MMSI'][f'{td[0]}']['Last_Updates'] = td[i]
+                    else:
+                        pass
+            return self.vessel_data_scrap(url,id,length + 1)
+
+    def Load_ALL_Vessel_Data(self,length=0):
+        """
+        loading the vessels data in data dictonary with recursion for faster extraction
+        """
+        if length == len(self.ids):
+            return self.data
+        else:
+            if self.data['ids'][f'{self.ids[length]}']['Vessels'] == None:
+                pass
+            else: 
+                url = self.data['ids'][f'{self.ids[length]}']['Vessels']['Status_URL']
+                self.vessel_data_scrap(url, self.ids[length])
+            return self.Load_ALL_Vessel_Data(length+1)
+
+    def run(self):
+        self.scrape_data() ## Extract
+        self.Load_ALL_Vessel_Data() ## Transform
+        with open('Submission.json','w') as output_file: ## Load (can be done over database(MongoDB, Mysql, etc) but for now just save in local file called submission.json )
```

Comment:
Using config dictionaries to specify the paths from where data is to be loaded and saved

Reply:
Hello,
Please check the updated commit i did in which i added one more function to specify the save destination of scraped data. here I integrate mongodb client as well so we can upload data directly there as well.
Thank you,
Kartik Arora